File: app/models/raw_gift.py
from app.extensions import db


class RawGift(db.Model):
    __tablename__ = "raw_gifts"

    id = db.Column(db.Integer, primary_key=True)
    snapshot_id = db.Column(db.Integer, db.ForeignKey("snapshots.id", ondelete="CASCADE"), nullable=False)
    department = db.Column(db.String(50), nullable=False)
    primary_addressee = db.Column(db.String(255))
    appeal_id = db.Column(db.String(255))
    split_amount = db.Column(db.Numeric(12, 2))
    fund_description = db.Column(db.String(255))
    gift_id = db.Column(db.Integer)
    gift_type = db.Column(db.String(100))
    gift_reference = db.Column(db.String(255))
    gift_date = db.Column(db.Date)
    extra_field = db.Column(db.String(255))

    def __repr__(self):
        return f"<RawGift {self.department}/{self.gift_id}>"

File: src/models/rawGift.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  return sequelize.define('RawGift', {
    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    snapshotId: { type: DataTypes.INTEGER, allowNull: false, field: 'snapshot_id' },
    department: { type: DataTypes.STRING(50), allowNull: false },
    primaryAddressee: { type: DataTypes.STRING(255), field: 'primary_addressee' },
    appealId: { type: DataTypes.STRING(255), field: 'appeal_id' },
    splitAmount: { type: DataTypes.DECIMAL(12, 2), field: 'split_amount' },
    fundDescription: { type: DataTypes.STRING(255), field: 'fund_description' },
    giftId: { type: DataTypes.INTEGER, field: 'gift_id' },
    giftType: { type: DataTypes.STRING(100), field: 'gift_type' },
    giftReference: { type: DataTypes.STRING(255), field: 'gift_reference' },
    giftDate: { type: DataTypes.DATEONLY, field: 'gift_date' },
    extraField: { type: DataTypes.STRING(255), field: 'extra_field' },
  }, {
    tableName: 'raw_gifts',
    timestamps: false,
  });
};
